frontend/src/app/feature/room/room.component.ts

```diff
@@ -39,131 +48,126 @@ export class RoomComponent extends WithWebSocket implements OnInit, OnDestroy {
-          this.room.pushMessage(`${message.data.name} already connected`);
+        if (!this.isMyOwnMessage(message)) {
+          const joinedPlayerId = message.senderId;
+          this.playerService.getPlayer(joinedPlayerId).subscribe(p => {
+              this.addPlayerTab(Player.newPlayer(p));
+              this.pushMessageToChat(`${p.name} joined room`);
+            });
         }
         break;
 
       case WsRoomMessageType.CHAT_MESSAGE:
-        this.room.pushMessage(message.data, this.room.getPlayerById(message.senderId));
+        const sender = this.getPlayerByUser(message.senderId);
+        this.pushMessageToChat(message.data, sender);
         break;
 
       case WsRoomMessageType.DISCONNECT:
-        this.room.pushMessage(`${this.room.getPlayerById(message.senderId).name} disconnected`);
-        this.room.getPlayerById(message.senderId).connected = false;
+        this.pushMessageToChat(`${this.getPlayerByUser(message.senderId).name} disconnected`);
         break;
 
       case WsRoomMessageType.INVENTORY:
         const inventory: Inventory = message.data;
         const items = inventory.items.reduce((acc, i) => `${acc}<br/> name: ${i.name} description: ${i.description}`, '');
-        this.room.pushMessage(`${this.room.getPlayerById(message.senderId).name}'s items :<br/>${items}`);
+        this.pushMessageToChat(`${this.getPlayerByUser(message.senderId).name}'s items :<br/>${items}`);
 
-        if (this.you.id !== message.senderId) {
-          this.room.players.get(message.senderId).inventory = message.data;
+        if (!this.isMyOwnMessage(message)) {
+          this.room.players.find(p => p.userId === this.currentPlayer.userId).inventory = message.data;
         }
         break;
     }
 
     setTimeout(() => this.chatbox.nativeElement.scrollTop = this.chatbox.nativeElement.scrollHeight);
   }
 
-  protected onOpen() {
-    this.sendMessage({ type: WsRoomMessageType.HI_I_AM_NEW_HERE, data: this.you });
-    this.room.pushMessage('Connected');
+  private isMyOwnMessage(message) {
+    return message.senderId === this.currentPlayer.id;
   }
 
-  protected onClose() {
-    this.sendMessage({ type: WsRoomMessageType.DISCONNECT });
+  private notifyAll = (params: WsMessageParam) => this.sendWsMessage({ roomId: this.room.id, senderId: this.currentPlayer.id, ...params });
+
+  private addPlayerTab(player: Player) {
+    if (this.room.players.indexOf(player) === -1) {
+      this.room.players.push(player);
+      this.playersSbj.next(this.room.players);
+    }
   }
 
-  private sendMessage = (params: WsMessageParam) => this.send({ roomId: this.room.id, ...params });
+  private getPlayerByUser(id: string) {
+    return this.room.players.find(p => p.id === id);
+  }
+
+  private pushMessageToChat(message: string, author: Player = this.systemPlayer, timestamp: string = new Date().toLocaleTimeString()) {
+    this.chatMessages.push(new RoomMessage(message, author, timestamp));
+  }
+
+  private throwIfEmpty(room) {
```

Comment:
Don't need to be a separate method

Reply:
I don't want to replace its call in the ngOnInit() with few lines. If you know how to have a single line with the body of throwIfEmpty() you can replace it. I moved this throwIfEmpty to a Util class since it may be useful in other places.